src/main/java/org/opensearch/sdk/ExtensionRouteHandlerFactory.java

```diff
@@ -0,0 +1,30 @@
+package org.opensearch.sdk;
+
+import org.opensearch.OpenSearchException;
+
+public class ExtensionRouteHandlerFactory {
+    private static ExtensionRouteHandlerFactory INSTANCE;
+
+    private String extensionShortName;
+
+    private ExtensionRouteHandlerFactory() { }
+
+    public static ExtensionRouteHandlerFactory getInstance() {
+        if(INSTANCE == null) {
+            INSTANCE = new ExtensionRouteHandlerFactory();
+        }
```

Comment:
Are we trying to create SINGLETON object for INSTANCE here? if yes is it multithreaded safe?

Reply:
Yes, the problem I was facing is that I needed the extensions shortName (abbreviation) inside the RestHandler classes like `RestHelloAction` in the sample extension. Instead of passing the value as a constructor arg or passing that extension runner into those classes, I thought it would be better to have singular place where its registered and can be used throughout.

Do you know how I can test if this is multi-threaded safe?